selfdrive/loggerd/omx_encoder.cc

```diff
@@ -448,7 +445,7 @@ int OmxEncoder::encode_frame(const uint8_t *y_ptr, const uint8_t *u_ptr, const u
   // this sometimes freezes... put it outside the encoder lock so we can still trigger rotates...
   // THIS IS A REALLY BAD IDEA, but apparently the race has to happen 30 times to trigger this
   //pthread_mutex_unlock(&this->lock);
-  OMX_BUFFERHEADERTYPE* in_buf = (OMX_BUFFERHEADERTYPE *)queue_pop(&this->free_in);
+  OMX_BUFFERHEADERTYPE* in_buf = this->free_in.pop();
```

Comment:
Or add a timeout parameter to try_pop,and call try_pop(timeout) here?

Reply:
Then we still need some early returns, which the caller of encode_frame needs to handle.